canonical-web-and-design/maas-ui
# feat(ui): add ability to create volume groups

## Done

- Added `BulkActions` component + types to house actions that can be performed on multiple storage devices at once
- Added `CreateVolumeGroup` component which is a form for creating a volume group from selected eligible devices

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the storage tab of a Ready or Allocated machine
- Change the storage layout to blank
- Without any checkboxes checked, hover over the disabled "Create volume group" button and see there is a tooltip
- Check the disk checkbox and then click the enabled "Create volume group" button and submit
- A new volume group should appear in Available, and the disk should have moved to Used (alongside an automatically created partition)
- Remove the volume group from the actions dropdown so the disk moves back to Available
- Create two disk partitions from the actions dropdown
- Check the checkbox for the disk and see that the "Create volume group" button stays disabled
- Uncheck the disk checkbox and check the two partition checkboxes
- The "Create volume group" button should be enabled
- Click it and see that the table shows the details of the two partitions and calculates the total size
- Submit the form and check that a volume group shows up in Available and the two partitions move to Used

## Fixes

Fixes canonical-web-and-design/MAAS-squad#2167

## Screenshot

![Screenshot_2020-12-11 fun-mammal maas storage bolla MAAS](https://user-images.githubusercontent.com/25733845/101861963-05607f80-3bbd-11eb-8e14-8523f2bb5789.png)
